### ISSUE TITLE
Class Declaration: classes cannot implement an interface

### ISSUE THREAD

**Issue by staycoolcall911** [winglang/wing#2111]
### I tried this

Currently we support `interface` `impl` only by `resource`s, not `class`es.
I tried this:
```ts (wing)
interface IAnimal {
  eat();
}

class Dog impl IAnimal {
  init(){}
  eat() {
    return;
  }
}

let z: IAnimal = new Dog();
```
If we replace `class` with `resource` in `Dog` definition it works, so it is just that `class`es still don't support implementing `interface`s. 

### I expected this:

To compile

### Instead, this happened

VSCode ext. crashed with this error:
```sh
[Info  - 11:11:49 AM] Connection to server got closed. Server will restart.
thread '<unnamed>' panicked at 'Unexpected class element node type method_definition || {Node method_definition (6, 2) - (8, 3)}', libs/wingc/src/parser.rs:557:21
stack backtrace:
wasm://wasm/0098ec7e:1


RuntimeError: unreachable
    at std::panicking::rust_panic_with_hook::h911ba172265603df (wasm://wasm/0098ec7e:wasm-function[220]:0xc8997)
    at core::panicking::panic_fmt::h6958314cbbe045b2 (wasm://wasm/0098ec7e:wasm-function[1432]:0x171d17)
    at wingc::parser::Parser::build_class_statement::h93ea20d069184bdb (wasm://wasm/0098ec7e:wasm-function[27]:0x3a3a9)
    at <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::try_fold::ha040a88cd775310e (wasm://wasm/0098ec7e:wasm-function[633]:0x13ac2c)
    at <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter::h42c603eb83c62d86 (wasm://wasm/0098ec7e:wasm-function[428]:0x1146ef)
    at wingc::parser::Parser::wingit::h8c70ecaf2a03c9c5 (wasm://wasm/0098ec7e:wasm-function[135]:0xa7a2f)
    at wingc::lsp::sync::partial_compile::h04133978863d627e (wasm://wasm/0098ec7e:wasm-function[150]:0xb0586)
    at wingc_on_did_open_text_document (wasm://wasm/0098ec7e:wasm-function[629]:0x138945)
    at wingc_on_did_open_text_document.command_export (wasm://wasm/0098ec7e:wasm-function[1574]:0x17a1f3)
    at Object.invoke (/Users/urib/.nvm/versions/node/v19.3.0/lib/node_modules/winglang/dist/wingc.js:89:37)
    at /Users/urib/.nvm/versions/node/v19.3.0/lib/node_modules/winglang/dist/commands/lsp.js:31:22
    at handleNotification (/Users/urib/.nvm/versions/node/v19.3.0/lib/node_modules/winglang/node_modules/vscode-jsonrpc/lib/common/connection.js:640:25)
    at handleMessage (/Users/urib/.nvm/versions/node/v19.3.0/lib/node_modules/winglang/node_modules/vscode-jsonrpc/lib/common/connection.js:342:13)
    at processMessageQueue (/Users/urib/.nvm/versions/node/v19.3.0/lib/node_modules/winglang/node_modules/vscode-jsonrpc/lib/common/connection.js:362:17)
    at Immediate.<anonymous> (/Users/urib/.nvm/versions/node/v19.3.0/lib/node_modules/winglang/node_modules/vscode-jsonrpc/lib/common/connection.js:334:13)
    at process.processImmediate (node:internal/timers:471:21)

Node.js v19.3.0
```

### Component

Compiler

### Wing Version

0.11.7

### Node.js Version

19.3.0

### Platform(s)

MacOS

### Anything else?

This is a TODO we know about - it's just a matter of priorities for our team. We will need to complete this in case we see use cases for this, but for now and given the change we are planning to make in this area with `resource interface` (see #1961) it's ok to hold. 

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by ekeren** [comment]
@staycoolcall911 following the team meeting a couple of minutes ago, please check that this issue is still an issue

**Comment by staycoolcall911** [comment]
This was fixed as part of #2270
